src/index.js

```diff
@@ -0,0 +1,118 @@
+// ========= CHANGE TEMP =========
+
+const state = {
+  temp: 42,
+};
+
+const increaseTemp = (event) => {
+  state.temp += 1;
+  const tempDisplayed = document.querySelector('#temp-value');
+  tempDisplayed.textContent = state.temp;
+  if (state.temp >= 80) {
+    tempDisplayed.style.color = 'red';
+  } else if (state.temp >= 70) {
+    tempDisplayed.style.color = 'pink';
+  } else if (state.temp >= 60) {
+    tempDisplayed.style.color = 'orange';
+  } else if (state.temp >= 50) {
+    tempDisplayed.style.color = 'green';
+  } else if (state.temp <= 49) {
+    tempDisplayed.style.color = 'blue';
+  }
+  changeLandscape();
+};
+
+const decreaseTemp = (event) => {
+  state.temp -= 1;
+  const tempDisplayed = document.querySelector('#temp-value');
+  tempDisplayed.textContent = state.temp;
+  if (state.temp >= 80) {
+    tempDisplayed.style.color = 'red';
+  } else if (state.temp >= 70) {
+    tempDisplayed.style.color = 'pink';
+  } else if (state.temp >= 60) {
+    tempDisplayed.style.color = 'orange';
+  } else if (state.temp >= 50) {
+    tempDisplayed.style.color = 'green';
+  } else if (state.temp <= 49) {
+    tempDisplayed.style.color = 'blue';
+  }
+  changeLandscape();
+};
+
+// ========= CHANGE CITY =========
+
+const changeCityName = (event) => {
+  const inputNewCity = document.getElementById('cityNameInputField').value;
+  const cityDisplayed = document.querySelector('#header-City-Name');
+  cityDisplayed.textContent = inputNewCity;
+};
+
+const resetCity = (event) => {
+  const inputDiffCity = document.getElementById('cityNameInputField');
+  cityNameInputField.value = 'Denver';
+  changeCityName();
+};
+
+// ========= CHANGE SKY =========
+
+const changeSky = (event) => {
+  const selectNewSky = document.getElementById('dropdown-sky').value;
+  const skyDisplayArea = document.getElementById('sky-image');
+
+  if (selectNewSky === 'Snowy') {
+    skyDisplayArea.textContent = '🌨❄️❄️❄️🌨❄️❄️🌨❄️❄️❄️🌨❄️❄️🌨';
+  } else if (selectNewSky === 'Cloudy') {
+    skyDisplayArea.textContent = '☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️';
+  } else if (selectNewSky === 'Rainy') {
+    skyDisplayArea.textContent = '🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧';
+  } else if (selectNewSky === 'Sunny') {
+    skyDisplayArea.textContent = '☁️ ☁️ ☁️ ☀️ ☁️ ☁️';
+  }
+};
```

Comment:
Consider the above logic and how similar all the branches are. We look for whether the selected sky matches a certain value, and then we set the text to display for the sky. What if we had a list of records that we could iterate through to find the new sky value. We could set up something like
```js
  const skyRecords = [
    { skyValue: "Sunny", display: "☁️ ☁️ ☁️ ☀️ ☁️ ☁️"},
    { skyValue: "Rainy", display: "🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧"},
    ...
  ];
```
if we wanted to iterate through, or something like
```js
  const skyRecords = {
     "Sunny": "☁️ ☁️ ☁️ ☀️ ☁️ ☁️",
     "Rainy": "🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧",
    ...
  };
```
for direct lookups (this wouldn't work if we tried to apply this to the equivalent updates for temperature ranges, which would work better with an iterative comparison).

All we need to do is look up the new display string and set it. Writing our code to be more data-driven would potentially allow us to populate these lists from a web service, allowing for new values without needing to modify the code!